zilliqa/src/api/eth.rs

```diff
@@ -292,10 +292,24 @@ fn get_transaction_count(params: Params, node: &Arc<Mutex<Node>>) -> Result<Stri
     expect_end_of_params(&mut params, 3, 3)?;
 
     let node = node.lock().unwrap();
+
     let block = node.get_block(block_id)?;
     let block = build_errored_response_for_missing_block(block_id, block)?;
 
-    Ok(node.get_state(&block)?.get_account(address)?.nonce.to_hex())
+    let nonce = node.get_state(&block)?.get_account(address)?.nonce;
+
+    if matches!(block_id, BlockId::Number(BlockNumberOrTag::Pending)) {
+        let pending_transaction_count = node
+            .txpool_content()
```

Comment:
I don't believe it would be very beneficial as there would be a load of duplicate code, with almost the same effect and some computation needs to happen in the `Consensus` anyway to ensure nonces are not too low. I have proposed a change moving the computation to `Consensus`. If you think that it should still be in `TransactionPool`, let me know and I will move all I can there.